docs/modules/ROOT/pages/erc721.adoc

```diff
@@ -1,772 +1,277 @@
-To properly interpret a URI from ERC721, simply trim the null bytes and decode the remaining bits as an ASCII string.
-For example:
-
-[,python]
-----
-# HELPER METHODS
-def str_to_felt(text):
-    b_text = bytes(text, 'ascii')
-    return int.from_bytes(b_text, "big")
-
-def felt_to_str(felt):
-    b_felt = felt.to_bytes(31, "big")
-    return b_felt.decode()
-
-token_id = uint(1)
-sample_uri = str_to_felt('mock://mytoken')
-
-await signer.send_transaction(
-    account, erc721.contract_address, 'setTokenURI', [
-        *token_id, sample_uri]
-)
-
-felt_uri = await erc721.tokenURI(first_token_id).call()
-string_uri = felt_to_str(felt_uri)
-----
-
-=== ERC721Received
-
-In order to be sure a contract can safely accept ERC721 tokens, said contract must implement the `ERC721_Receiver` interface (as expressed in the EIP721 specification).
-Methods such as `safeTransferFrom` and `safeMint` call the recipient contract's `onERC721Received` method.
-If the contract fails to return the correct magic value, the transaction fails.
-
-StarkNet contracts that support safe transfers, however, must also support xref:introspection.adoc#erc165[ERC165] and include `supportsInterface` as proposed in https://github.com/OpenZeppelin/cairo-contracts/discussions/100[#100].
-`safeTransferFrom` requires a means of differentiating between account and non-account contracts.
-Currently, StarkNet does not support error handling from the contract level;
-therefore, the current ERC721 implementation requires that all contracts that support safe ERC721 transfers (both accounts and non-accounts) include the `supportsInterface` method.
-Further, `supportsInterface` should return `TRUE` if the recipient contract supports the `IERC721Receiver` magic value `0x150b7a02` (which invokes `onERC721Received`).
-If the recipient contract supports the `IAccount` magic value `0x50b70dcb`, `supportsInterface` should return `TRUE`.
-Otherwise, `safeTransferFrom` should fail.
-
-==== IERC721Receiver
-
-Interface for any contract that wants to support safeTransfers from ERC721 asset contracts.
-
-[,cairo]
-----
-@contract_interface
-namespace IERC721Receiver {
-    func onERC721Received(
-        operator: felt, from_: felt, tokenId: Uint256, data_len: felt data: felt*) -> (selector: felt) {
+:mint-api: xref:api/erc721.adoc#ERC721-_mint[_mint]
+
+Using Contracts for Cairo, constructing an ERC721 contract requires integrating both `ERC721Component` and `SRC5Component`.
+The contract should also set up the constructor to initialize the token's name, symbol, and interface support.
+Here's an example of a basic contract:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyNFT {
+    use openzeppelin::introspection::src5::SRC5Component;
+    use openzeppelin::token::erc721::ERC721Component;
+    use starknet::ContractAddress;
+
+    component!(path: SRC5Component, storage: src5, event: SRC5Event);
+    component!(path: ERC721Component, storage: erc721, event: ERC721Event);
```

Comment:
I think we should swap the order (this applies to the impls below as well). This is compliant with the account preset too.